addons/crm/models/crm_lead.py

```diff
@@ -856,6 +859,19 @@ def action_snooze(self):
             })
         return True
 
+    def action_reschedule_meeting(self):
+        self.ensure_one()
+        next_calendar_event = self.activity_ids.filtered(lambda activity: activity.user_id == self.env.user)[:1].calendar_event_id
```

Comment:
Why [:1] ? Why not all future events for example and going on a calendar view, or a list view ?

Reply:
I implemented this with regards to how the Snooze 7d works. It only affects the next activity. Then, if by rescheduling/snoozing it, another activity becomes the earliest, the snoozing/rescheduling will apply on that activity.

Also, the specification on the task was as follow :
> If it's an activity type that creates a calendar event, instead of snooze display "fa-calendar Reschedule" and show this modal: https://tinyurl.com/y2qz9avo

I don't mind doing otherwise if it's better !

Reply:
Well snooze does not redirect to another view. It was then chosen to do it only for the first upcoming one. Here if you redirect to another view, we could include all event-based activities... maybe to check with LNA.